File: examples/too_many_cooks_vscode_extension/src/test/suite/configuration.test.ts
/**
 * Configuration Tests
 * Verifies configuration settings work correctly.
 */

import * as assert from 'assert';
import * as vscode from 'vscode';
import { waitForExtensionActivation } from '../test-helpers';

suite('Configuration', () => {
  suiteSetup(async () => {
    await waitForExtensionActivation();
  });

  test('serverPath configuration exists', () => {
    const config = vscode.workspace.getConfiguration('tooManyCooks');
    const serverPath = config.get<string>('serverPath');
    assert.ok(serverPath !== undefined, 'serverPath config should exist');
  });

  test('autoConnect configuration exists', () => {
    const config = vscode.workspace.getConfiguration('tooManyCooks');
    const autoConnect = config.get<boolean>('autoConnect');
    assert.ok(autoConnect !== undefined, 'autoConnect config should exist');
  });

  test('autoConnect defaults to true', () => {
    const config = vscode.workspace.getConfiguration('tooManyCooks');
    const autoConnect = config.get<boolean>('autoConnect');
    // Default is true according to package.json
    assert.strictEqual(autoConnect, true);
  });

  test('serverPath can be updated', async () => {
    const config = vscode.workspace.getConfiguration('tooManyCooks');
    const testPath = '/test/path/to/server.js';

    await config.update('serverPath', testPath, vscode.ConfigurationTarget.Global);

    const updatedConfig = vscode.workspace.getConfiguration('tooManyCooks');
    const serverPath = updatedConfig.get<string>('serverPath');
    assert.strictEqual(serverPath, testPath);

    // Reset
    await config.update('serverPath', '', vscode.ConfigurationTarget.Global);
  });
});

File: examples/too_many_cooks_vscode_extension_dart/test/suite/configuration_test.dart
/// Configuration Tests
/// Verifies configuration settings work correctly.
library;

import 'dart:js_interop';

import 'package:dart_node_vsix/dart_node_vsix.dart';

import 'test_helpers.dart';

@JS('console.log')
external void _log(String msg);

void main() {
  _log('[CONFIGURATION TEST] main() called');

  // Ensure any dialog mocks from previous tests are restored
  restoreDialogMocks();

  suite('Configuration', syncTest(() {
    suiteSetup(asyncTest(() async {
      _log('[CONFIG] suiteSetup - waiting for extension activation');
      await waitForExtensionActivation();
    }));

    test('autoConnect configuration exists', syncTest(() {
      _log('[CONFIG] Running: autoConnect configuration exists');
      final config = vscode.workspace.getConfiguration('tooManyCooks');
      final autoConnect = config.get<JSBoolean>('autoConnect');
      assertOk(
        !autoConnect.isUndefinedOrNull,
        'autoConnect config should exist',
      );
      _log('[CONFIG] PASSED: autoConnect configuration exists');
    }));

    test('autoConnect defaults to true', syncTest(() {
      _log('[CONFIG] Running: autoConnect defaults to true');
      final config = vscode.workspace.getConfiguration('tooManyCooks');
      final autoConnect = config.get<JSBoolean>('autoConnect');
      // Default is true according to package.json
      assertEqual(autoConnect?.toDart, true);
      _log('[CONFIG] PASSED: autoConnect defaults to true');
    }));
  }));

  _log('[CONFIGURATION TEST] main() completed');
}
